dao/progress.go

```diff
@@ -0,0 +1,24 @@
+package dao
+
+import (
+	"TUM-Live/model"
+	"gorm.io/gorm/clause"
+)
+
+func SaveProgress(progress float64, userID uint, streamID uint) {
+	// Update columns to new value on `id` conflict
+	DB.Clauses(clause.OnConflict{
+		Columns:   []clause.Column{{Name: "stream_id"}, {Name: "user_id"}}, // key column
+		DoUpdates: clause.AssignmentColumns([]string{"progress"}), // column needed to be updated
+	}).Create(&model.StreamProgress{
+		Progress: progress,
+		StreamID: streamID,
+		UserID:   userID,
+	})
+}
+
+func LoadProgress(userID uint, streamID uint) float64 {
```

Comment:
For good measure this could return an error so the `/w/GBS/123` handler can take care of setting the progress to 0 (if not found) or logging it in case of any other error. See https://gorm.io/docs/error_handling.html#ErrRecordNotFound

Reply:
Good idea, this should improve robustness. :+1: